### PR TITLE
Break text on whole words

### PR DESCRIPTION
**Pull request by themadcreator**
Modify logic of `Wrapper` to better wrap whole words when `wrapper.allowBreakingWords` is set to `false`.
    
Also, include punctuation in tokens to prevent single character punctuation from wrapping to its own line.
    
Update preview and unit tests to confirm new behavior.

#### New wrap example
Given the string and the available space (`|---|`):
```
"Hello, world!"
|-------|
```
Previously this would wrap like so:
```
|-------|
Hello, w-
orld!
```

With this change, we now wrap like so:
```
|-------|
Hello,
world!
```

The old behavior can still be used by setting `wrapper.allowBreakingWords(true);`

#### Last resort breaking

Notably, a single word that does not fit in one line may still be broken and hyphenated. like so:

```
"A very longwordthatdoesntfit"
|------|
A very
longwor-
dthatdo-
esntfit
```

#### Punctuation

Additionally, we now consider punctuation to be part of the previous token. This prevents single character punctuation such as "," or "." from being wrapped to its own line.

Previous:
```
|-----|
A comma
, too.
```

New
```
|-----|
A
comma,
too.
```

### CODE REVIEW

File: src/wrappers/wrapper.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export class Wrapper {
   constructor() {
     this.maxLines(Infinity);
     this.textTrimming("ellipsis");
-    this.allowBreakingWords(true);
+    this.allowBreakingWords(false);
```
Comment (hellochar) [inline src/wrappers/wrapper.ts]: We shouldn't change the default since it's a significant breaking change.

Comment (themadcreator) [reply src/wrappers/wrapper.ts]: Can we at least set this as this as the default when we update plottable? Line-wrapping naively in the middle of a word is pretty ugly.

Comment (hellochar) [reply src/wrappers/wrapper.ts]: Yeah we can turn it on in Plottable by default

Comment (themadcreator) [reply src/wrappers/wrapper.ts]: Yeah, let's major rev.

**Pull request merged**